extensions/arc/src/utils.ts

```diff
@@ -21,6 +21,16 @@ export function getSpaceTitle(space: Space) {
   return space.title || `Space ${space.id}`;
 }
 
+export function findSpaceInSpaces(spaceId: string, spaces: Space[]): string | undefined {
+  for (const s of spaces) {
+    const title = getSpaceTitle(s);
+    if (title.toLowerCase() === spaceId.toLowerCase() || s.id.toString() === spaceId) {
+      return title;
+    }
+  }
+  return undefined;
+}
```

Comment:
```suggestion
export function findSpaceInSpaces(spaceId: string, spaces: Space[]): string | undefined {
  const space = spaces.find(s => 
     getSpaceTitle(s).toLowerCase() === spaceId.toLowerCase() || 
     s.id.toString() === spaceId
  );
  return space && getSpaceTitle(space);
}
```
Could also be written something like that, not sure if it works since though since I didn't test it 🙂

Reply:
Oh, definitely. I've DM'd you on Slack.